akka-sample-persistence-dc-java/src/main/java/sample/persistence/multidc/ThumbsUpCounter.java

```diff
@@ -1,170 +1,140 @@
     }
-  }
 
-  public static class GetCount implements Command {
-    public final String resourceId;
+    public static class GiveThumbsUp implements Command {
+        public final String resourceId;
+        public final String userId;
+        public final ActorRef<Long> replyTo;
 
-    public GetCount(String resourceId) {
-      this.resourceId = resourceId;
+        public GiveThumbsUp(String resourceId, String userId, ActorRef<Long> replyTo) {
+            this.resourceId = resourceId;
+            this.userId = userId;
+            this.replyTo = replyTo;
+        }
     }
 
-    public String getResourceId() {
-      return resourceId;
+    public static class GetCount implements Command {
+        public final String resourceId;
+        public final ActorRef<Long> replyTo;
+
+        public GetCount(String resourceId, ActorRef<Long> replyTo) {
+            this.resourceId = resourceId;
+            this.replyTo = replyTo;
+        }
     }
-  }
 
-  public static class GetUsers implements Command {
-    public final String resourceId;
+    public static class GetUsers implements Command {
+        public final String resourceId;
+        public final ActorRef<State> replyTo;
 
-    public GetUsers(String resourceId) {
-      this.resourceId = resourceId;
-    }
+        public GetUsers(String resourceId, ActorRef<State> replyTo) {
+            this.resourceId = resourceId;
+            this.replyTo = replyTo;
+        }
 
-    public String getResourceId() {
-      return resourceId;
+        public String getResourceId() {
+            return resourceId;
+        }
     }
-  }
 
 
-  interface Event {}
+    interface Event extends CborSerializer {
+    }
 
-  public static class GaveThumbsUp implements Event {
-    public final String userId;
+    public static class GaveThumbsUp implements Event {
+        public final String userId;
 
-    public GaveThumbsUp(String userId) {
-      this.userId = userId;
+        @JsonCreator
+        public GaveThumbsUp(String userId) {
+            this.userId = userId;
+        }
     }
-  }
 
-  public static class State {
-    public final Set<String> users;
+    public static class State implements CborSerializer {
+        public final Set<String> users;
 
-    public State() {
-      this.users = Collections.emptySet();
-    }
+        public State() {
+            this.users = Collections.emptySet();
+        }
 
-    State(Set<String> users) {
-      this.users = Collections.unmodifiableSet(users);
-    }
+        State(Set<String> users) {
+            this.users = Collections.unmodifiableSet(users);
+        }
 
-    public State add(String userId) {
-      // defensive copy for thread safety
-      Set<String> newUsers = new HashSet<>(users);
-      newUsers.add(userId);
-      return new State(newUsers);
+        public State add(String userId) {
+            // defensive copy for thread safety
+            Set<String> newUsers = new HashSet<>(users);
```

Comment:
`State` can be mutable (as long as it's not sent as message, so need to fight it